src/main/java/com/twcable/grabbit/tools/monitor/PollingJobMonitor.java

```diff
@@ -0,0 +1,343 @@
+                hasRunningJobs = poll();
+            }
+
+            monitoringEventSubscriber.onNext(new EndMonitoringEvent(startTime, Instant.now()));
+            monitoringEventSubscriber.onComplete();
+        }
+
+
+        private boolean poll() {
+            monitoringEventSubscriber.onNext(new StartMonitoringEvent());
+
+            final Collection<JobStatus> jobResults = jobStatusesForRunningJobs(monitoringEventSubscriber);
+
+            publishCompletionReports(monitoringEventSubscriber, jobResults);
+
+            final boolean hasRunningJobs = jobResults.stream().anyMatch(this::isRunning);
+
+            if (hasRunningJobs) {
+                sleep(monitoringEventSubscriber);
+            }
+            return hasRunningJobs;
+        }
+
+
+        private boolean isRunning(JobStatus jobStatus) {
+            return jobStatus.state() == RUNNING;
+        }
+
+
+        private void publishCompletionReports(Subscriber<MonitoringEvent> subscriber,
+                                              Collection<JobStatus> jobResults) {
+            subscriber.onNext(new CompletedJobsMonitoringEvent(jobsOfState(jobResults, COMPLETED)));
+            subscriber.onNext(new FailedJobsMonitoringEvent(jobsOfState(jobResults, FAILED)));
+        }
+
+
+        private Collection<@NonNull JobStatus> jobsOfState(Collection<JobStatus> jobResults, JobState jobState) {
+            return jobResults.stream().
+                filter(jobStatus -> jobStatus.state() == jobState).
+                collect(Collectors.<@NonNull JobStatus>toList());
+        }
+
+
+        private Collection<JobStatus> jobStatusesForRunningJobs(Subscriber<MonitoringEvent> subscriber) {
+            return jobStatusCache.entries().stream().
+                filter(entry -> entry.state == RUNNING).
+                map(entry -> {
+                    subscriber.onNext(new PollingMonitoringEvent(entry.location, entry.jobId));
+                    final JobStatus jobStatus = jobStatusPoller.pollJobStatus(entry.location, entry.jobId);
+                    jobStatusCache.put(entry.location, jobStatus.jobExecutionId, jobStatus.state());
+                    return jobStatus;
+                }).
+                collect(Collectors.<@NonNull JobStatus>toList());
+        }
+
+
+        private void sleep(Subscriber<MonitoringEvent> subscriber) {
+            subscriber.onNext(new SleepMonitoringEvent(sleepMs));
+            try {
+                Thread.sleep(sleepMs);
+            }
+            catch (InterruptedException e) {
+                throw softened(e);
+            }
+        }
+    }
+
+
+    /**
+     * Simple namespace for builder classes/interfaces.
+     *
+     * The builder uses a number of trivial interfaces to ensure that the type system enforces that it is impossible
+     * to call {@link Build#build()} in an invalid state.
+     */
+    @SuppressWarnings({"unused", "checkstyle:JavadocType", "PMD.AccessorClassGeneration"})
+    public interface B {
```

Comment:
This is a lot of lines of code, but they're all trivial. As the comment says, it ensures that `PollingJobMonitor` can only be created in a consistent state, and simplifies its use.